02-oop-inheritance/script.js

```diff
@@ -0,0 +1,40 @@
+var movie1 = Movie('What a wonderful life', 1946, '2h 10min');
+var movie2 = Movie('The Shining', 1980, '2h 26min');
+var movie3 = Movie('Amadeus', 1984, '2h 40min');
+var movie4 = Movie('Back to the Future', 1985, '1h 56min');
+var movie5 = Movie('The Fountain', 2006, '1h 36min');
+
+function Movie(title, year, duration){
+	return {
+		title,
+		year,
+		duration
+	}
+}
+
+ 
+function doSomething(e) {
+    alert("Event is called: " + e.type);
+}
+
+var an_event = new CustomEvent("event");
+
+ 
+function EventEmmiter(){
+	var myEvent, func;
+	return {
+		myEvent, func, 
+		on: function(one_event, fnc){
+			myEvent = one_event;
+			func = fnc;
+			document.addEventListener(myEvent.type, func, false);
```

Comment:
It's not necessary to use javascript builtin event listeners, you can do it by attaching a list of event handlers. like:

```
var eventHandlers = [
 { 
   name: 'play', 
   callback: function() { 
     console.log('played') 
  },
 { 
   name: 'stop', 
   callback: function() { 
     console.log('stopped') 
  },
 }
]
```

On: allows us to add handler objects to the list. 
Emit: allows us to execute the specified event callback if exists.
Off: allows us to remove the handler object from the list if exists.